pkg/controller/master/backup/backup_target.go

```diff
@@ -111,6 +111,9 @@ func (h *TargetHandler) OnBackupTargetChange(key string, setting *harvesterv1.Se
 			return nil, err
 		}
 
+		// set configured flag, as s3 does
+		harvesterv1.SettingConfigured.SetError(setting, "", nil)
```

Comment:
We need to get a deep copy of the resource, change it then use the client to update it. Otherwise, we are updating the resource in the memory cache and it's not persistent in etcd.
FYI this is how we did this before: https://github.com/harvester/harvester/blob/v0.3/pkg/controller/master/backup/backupTaget.go#L95

Reply:
@gitlawr I think u already did in https://github.com/harvester/harvester/blob/f5bde7fa462c09d0513b8f370f71f2dd974f298c/pkg/controller/master/setting/handler.go#L65 so it is not needed?

Reply:
It only affects specific settings declared in https://github.com/harvester/harvester/blob/f5bde7fa462c09d0513b8f370f71f2dd974f298c/pkg/controller/master/setting/register.go#L47-L54